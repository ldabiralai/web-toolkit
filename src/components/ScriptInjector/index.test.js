import React from 'react';
import { mount } from 'enzyme';
import { ScriptInjector } from '../..';

describe('ScriptInjector', () => {
  describe('server side', () => {
    it('matches default snapshot', () => {
      expect(mount(<ScriptInjector isServer />)).toMatchSnapshot();
    });

    describe('props', () => {
      let wrapper;
      beforeEach(() => {
        wrapper = mount(
          <ScriptInjector
            id="scriptinjector"
            isServer
            async
            src="https://test/script.src"
            innerHTML={'inner html text <b />'}
          />
        );
      });

      it('async', () => {
        expect(wrapper.prop('async')).toBe(true);
      });

      it('src', () => {
        expect(wrapper.prop('src')).toBe('https://test/script.src');
      });

      it('innerHTML', () => {
        expect(wrapper.prop('innerHTML')).toBe('inner html text <b />');
      });
    });
  });

  describe('client side', () => {
    it('matches default snapshot', () => {
      const wrapper = mount(<ScriptInjector isServer={false} />);
      expect(wrapper).toMatchSnapshot();
      wrapper.unmount();
    });

    describe('injectionPlace', () => {
      let wrapper;

      afterAll(() => {
        wrapper.unmount();
      });

      it('HEAD by default', () => {
        wrapper = mount(<ScriptInjector id="scriptinjector" isServer={false} async src="https://test/script.src" />);
        expect(document.getElementById('scriptinjector').parentNode.tagName).toBe('HEAD');

        wrapper.unmount();
        expect(document.getElementById('scriptinjector')).toBe(null);
      });

      it('BODY', () => {
        wrapper = mount(
          <ScriptInjector id="scriptinjector" isServer={false} async src="https://test/script.src" injectPlace="body" />
        );
        expect(document.getElementById('scriptinjector').parentNode.tagName).toBe('BODY');

        wrapper.unmount();
        expect(document.getElementById('scriptinjector')).toBe(null);
      });
    });

    describe('script attributes and innerHTML', () => {
      let wrapper;
      const onLoadSpy = jest.fn();

      beforeEach(() => {
        wrapper = mount(
          <ScriptInjector
            id="scriptinjector"
            isServer={false}
            async
            src="https://test/script.src"
            innerHTML={'inner html text <b />'}
            onLoad={() => onLoadSpy(123)}
          />
        );
      });

      afterEach(() => {
        wrapper.unmount();
      });

      it('has src', () => {
        expect(document.getElementById('scriptinjector').src).toBe('https://test/script.src');
      });

      it('has async', () => {
        expect(document.getElementById('scriptinjector').async).toBe(true);
      });

      it('has innerHTML', () => {
        expect(document.getElementById('scriptinjector').innerHTML).toBe('inner html text <b />');
      });

      it('invokes onLoad', () => {
        wrapper.instance().script.onload();
        expect(onLoadSpy).toHaveBeenCalledWith(123);
      });

      it('does not inject same script id twice', () => {
        const oneMoreWrapper = mount(
          <ScriptInjector
            id="scriptinjector"
            isServer={false}
            async
            src="https://test/script.src"
            innerHTML={'inner html text <b />'}
            onLoad={() => onLoadSpy(123)}
          />
        );

        expect(document.querySelectorAll('script')).toHaveLength(1);
        oneMoreWrapper.unmount();
      });
    });
  });
});
