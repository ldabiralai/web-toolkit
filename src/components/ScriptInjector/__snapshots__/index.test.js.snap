// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptInjector client side matches default snapshot 1`] = `
<ScriptInjector
  async={false}
  injectPlace="head"
  innerHTML=""
  isServer={false}
  src=""
/>
`;

exports[`ScriptInjector server side matches default snapshot 1`] = `
<ScriptInjector
  async={false}
  injectPlace="head"
  innerHTML=""
  isServer={true}
  src=""
>
  <script
    async={false}
    src=""
  />
</ScriptInjector>
`;
